src/nomad_simulations/schema_packages/general.py

```diff
@@ -269,23 +276,28 @@ def get_composition_recurs(system: ModelSystem, atom_labels: list[str]) -> None:
                 to the atom indices stored in system.
             """
             subsystems = system.model_system
-            set_composition_formula(
-                system=system, subsystems=subsystems, atom_labels=atom_labels
-            )
+            set_composition_formula(system=system, subsystems=subsystems, labels=labels)
             if subsystems:
                 for subsystem in subsystems:
-                    get_composition_recurs(system=subsystem, atom_labels=atom_labels)
+                    get_composition_recurs(system=subsystem, labels=labels)
 
         # ! CG: system_parent.cell[0].particles_state instead of atoms_state!
-        atoms_state = (
-            system_parent.cell[0].atoms_state if system_parent.cell is not None else []
-        )
-        atom_labels = (
-            [atom.chemical_symbol for atom in atoms_state]
-            if atoms_state is not None
-            else []
-        )
-        get_composition_recurs(system=system_parent, atom_labels=atom_labels)
+        labels = []
+        if system_parent.cell is not None:
+            if system_parent.cell[0].name == 'AtomicCell':
```

Comment:
This is just a temporary solution to make sure the rest works. 

I think now that we can use some method of an archive section called m_xpath to grab the section by name of the parent class...see get_sibling_section() in utils.py

More generally I am also considering making an intermediate parent for AtomicCell and ParticleCell to share particle-specific quantities and functionalities that we don't want in cell...